client/web/src/site-admin/WebhookInformation.tsx

```diff
@@ -0,0 +1,60 @@
+import { FC, useState } from 'react'
+
+import { mdiContentCopy, mdiEye } from '@mdi/js'
+import classNames from 'classnames'
+import copy from 'copy-to-clipboard'
+
+import { Button, Icon, Input } from '@sourcegraph/wildcard'
+
+import { WebhookFields } from '../graphql-operations'
+
+import styles from './WebhookInformation.module.scss'
+
+export interface WebhookInformationProps {
+    webhook: WebhookFields
+}
+
+export const WebhookInformation: FC<WebhookInformationProps> = props => {
+    const { webhook } = props
+    const [secretShown, setSecretShown] = useState(false)
+
+    return (
+        <table className={classNames(styles.table, 'table')}>
+            <tbody>
+                <tr>
+                    <th>Code host</th>
+                    <td>{webhook.codeHostKind}</td>
+                </tr>
+                <tr>
+                    <th>URN</th>
+                    <td>{webhook.codeHostURN}</td>
+                </tr>
+                <tr>
+                    <th>Webhook endpoint</th>
+                    <td className={styles.contentCell}>
+                        {webhook.url}
+                        <Button size="sm" className={styles.copyButton} onClick={() => copy(webhook.url)}>
+                            <Icon svgPath={mdiContentCopy} inline={true} aria-label="Copy webhook endpoint" />
+                        </Button>
+                    </td>
+                </tr>
+                <tr>
+                    <th>Secret</th>
+                    <td className={styles.contentCell}>
+                        <Input
+                            name="secret"
+                            type={secretShown ? 'text' : 'password'}
+                            value={webhook.secret ?? ''}
```

Comment:
Might want to do so that the dot count doesn't match the secret length? Not sure.

```suggestion
                            value={secretShown ? webhook.secret ?? '': 'versecretvalue'}
```

Reply:
agree, nice thing to have, added it